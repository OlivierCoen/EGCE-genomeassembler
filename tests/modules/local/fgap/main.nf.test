nextflow_process {

    name "Test Process FGAP"
    script "modules/local/fgap/main.nf"
    process "FGAP"
    tag "fgap"

    test("No gap to close") {

        when {

            process {
                """
                input[0] = [
                    [id: "test"],
                    file( '$projectDir/tests/test_data/long_reads/a_thaliana/a_thaliana_ont_test.fa.gz', checkIfExists: true),
                    file( '$projectDir/tests/test_data/scaffolded/a_thaliana_ont_test_scaffolds_final_purged.fa.gz', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Example data provided by FGAP") {

        when {

            process {
                """
                input[0] = [
                    [id: "test"],
                    file( 'https://raw.githubusercontent.com/pirovc/fgap/master/sample_data/DATASET_ecoli_454.fasta', checkIfExists: true),
                    file( 'https://raw.githubusercontent.com/pirovc/fgap/master/sample_data/DRAFT_ecoli_hiseq454.fasta', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
