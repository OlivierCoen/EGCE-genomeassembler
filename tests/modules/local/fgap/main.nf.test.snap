{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test:md5,4282d8b3a08f17fba7a69bb97a380e47"
                        },
                        "a_thaliana_ont_test_scaffolds_final_purged.fa_gapclosed.fasta.gz:md5,6023275fd5003c458eb1d32a7a2a59b4"
                    ]
                ],
                "1": [
                    [
                        "FGAP",
                        "fgap",
                        "1.8.1"
                    ]
                ],
                "2": [
                    [
                        "FGAP",
                        "pigz",
                        "2.8"
                    ]
                ],
                "gapclosed_assemblies": [
                    [
                        {
                            "id": "test:md5,4282d8b3a08f17fba7a69bb97a380e47"
                        },
                        "a_thaliana_ont_test_scaffolds_final_purged.fa_gapclosed.fasta.gz:md5,6023275fd5003c458eb1d32a7a2a59b4"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-23T17:41:08.88986705"
    },
    "No gap to close": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "a_thaliana_ont_test_scaffolds_final_purged_gapclosed.fasta.gz:md5,6023275fd5003c458eb1d32a7a2a59b4"
                    ]
                ],
                "1": [
                    [
                        "FGAP",
                        "fgap",
                        "1.8.1"
                    ]
                ],
                "2": [
                    [
                        "FGAP",
                        "pigz",
                        "2.8"
                    ]
                ],
                "gapclosed_assemblies": [
                    [
                        {
                            "id": "test"
                        },
                        "a_thaliana_ont_test_scaffolds_final_purged_gapclosed.fasta.gz:md5,6023275fd5003c458eb1d32a7a2a59b4"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-23T18:30:51.912615555"
    },
    "Example data provided by FGAP": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "DRAFT_ecoli_hiseq454_gapclosed.fasta.gz:md5,60b991f00a53507e346dee76b3e1b04d"
                    ]
                ],
                "1": [
                    [
                        "FGAP",
                        "fgap",
                        "1.8.1"
                    ]
                ],
                "2": [
                    [
                        "FGAP",
                        "pigz",
                        "2.8"
                    ]
                ],
                "gapclosed_assemblies": [
                    [
                        {
                            "id": "test"
                        },
                        "DRAFT_ecoli_hiseq454_gapclosed.fasta.gz:md5,60b991f00a53507e346dee76b3e1b04d"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-23T18:31:26.282062012"
    }
}